drltrace/drltrace_frontend.cpp

```diff
@@ -0,0 +1,314 @@
+starting with \"_\".", "Ignores library routine names starting with \"_\".");
+static droption_t<bool> op_help
+(DROPTION_SCOPE_FRONTEND, "help", false, "Print this message.", "Print this message");
+static droption_t<bool> op_version
+(DROPTION_SCOPE_FRONTEND, "version", 0, "Print version number.", "Print version number.");
+static droption_t<unsigned int> op_verbose
+(DROPTION_SCOPE_ALL, "verbose", 1, "Change verbosity (default 1).",
+"Change verbosity (default 1).");
+static droption_t<std::string> op_ltracelib_ops
+(DROPTION_SCOPE_CLIENT, "ltracelib_ops",
+ DROPTION_FLAG_SWEEP | DROPTION_FLAG_ACCUMULATE | DROPTION_FLAG_INTERNAL,
+ "", "(For internal use: sweeps up drltracelib options)",
+ "This is an internal option that sweeps up other options to pass to the drltracelib.");
+
+/* check that drltracelib.dll, dynamorio.dll and target executable exist */
+static void
+check_input_files(const char *targer_app_full_name, char *dr_root, char *drltrace_path) {
+    bool result = false;
+
+    /* check that the target applicaton exist */
+    if (targer_app_full_name[0] == '\0')
+        DRLTRACE_ERROR("target application is not specified");
+    if (drfront_access(targer_app_full_name, DRFRONT_READ, &result) != DRFRONT_SUCCESS)
+        DRLTRACE_ERROR("cannot find target application at %s", targer_app_full_name);
+    if (!result)
+        DRLTRACE_ERROR("cannot open target application for read at %s",
+                       targer_app_full_name);
+
+    /* check that dynamorio's root dir exist and accessible */
+    if (drfront_access(dr_root, DRFRONT_READ, &result) != DRFRONT_SUCCESS)
+        DRLTRACE_ERROR("cannot find DynamoRIO's root dir at %s", dr_root);
+    if (!result)
+        DRLTRACE_ERROR("cannot open DynamoRIO's root dir for read at %s", dr_root);
+
+    /* check that drfrontendlib exist */
+    if (drfront_access(drltrace_path, DRFRONT_READ, &result) != DRFRONT_SUCCESS)
+        DRLTRACE_ERROR("cannot find drltracelib at %s", drltrace_path);
+    if (!result)
+        DRLTRACE_ERROR("cannot open drltracelib for read at %s", drltrace_path);
+}
+
+static void
+print_version() {
+#if defined(BUILD_NUMBER) && defined(VERSION_NUMBER)
+    printf("drltrace version %s -- build %d\n", STRINGIFY(VERSION_NUMBER), BUILD_NUMBER);
+#elif defined(BUILD_NUMBER)
+    printf(TOOLNAME" custom build %d -- %s\n", BUILD_NUMBER, __DATE__);
+#else
+    printf(TOOLNAME" custom build -- %s, %s\n", __DATE__, __TIME__);
+#endif
+    exit(0);
+}
+
+static void
+configure_application(char *app_name, char **app_argv, void **inject_data,
+                      const char *dr_root, const char *lib_path)
+{
+    bool is_debug = false;
+#ifdef DEBUG
+    is_debug = true;
+#endif
+    int errcode;
+    char *process;
+    process_id_t pid;
+    char dr_option[MAX_DR_CMDLINE];
+    dr_option[0] = '\0';
+
+    if (op_no_follow_children.get_value() == true)
+        _snprintf(dr_option, strlen("-no_follow_children") + 1, "-no_follow_children");
```

Comment:
Use dr_snprintf: _snprintf is Windows-only.  I would expect this to not compile on Linux.  Did it?

I see more instances below.